backend/node/genvm/base.py

```diff
@@ -13,6 +13,16 @@
 from backend.node.genvm.code_enforcement import code_enforcement_check
 
 
+class ContractRunner:
+    def __init__(self, mode: str, node_config: dict):
+        self.mode = mode
```

Comment:
I'm still not familiar entirely with the codebase, so just to learn: Would it be better to have 2 types of `ContractRunners`, one for validators and another one for leaders?

Reply:
I don't think so :) The ContractRunner supports the execution of the IC code. The mode is just a property to understand how to execute some parts of the code and how to process the output of the execution.